xmtp_mls/src/subscriptions.rs

```diff
@@ -167,48 +167,50 @@ where
                 }
             });
 
-        Ok(Box::pin(futures::stream::select(stream, event_queue)))
+        Ok(futures::stream::select(stream, event_queue))
     }
 
-    #[tracing::instrument(skip(self, group_id_to_info))]
+    // #[tracing::instrument(skip(self, group_id_to_info))]
     pub(crate) async fn stream_messages(
-        self: Arc<Self>,
-        group_id_to_info: HashMap<Vec<u8>, MessagesStreamInfo>,
-    ) -> Result<Pin<Box<dyn Stream<Item = StoredGroupMessage> + Send>>, ClientError> {
+        &self,
+        group_id_to_info: Arc<HashMap<Vec<u8>, MessagesStreamInfo>>,
+    ) -> Result<impl Stream<Item = StoredGroupMessage> + '_, ClientError>
+    where
+        ApiClient: 'static,
+    {
         let filters: Vec<GroupFilter> = group_id_to_info
             .iter()
             .map(|(group_id, info)| GroupFilter::new(group_id.clone(), Some(info.cursor)))
             .collect();
+
         let messages_subscription = self.api_client.subscribe_group_messages(filters).await?;
 
         let stream = messages_subscription
             .map(move |res| {
-                let context = self.context.clone();
-                let client = self.clone();
-
-                let group_id_to_info = group_id_to_info.clone();
+                let group_info = group_id_to_info.clone();
```

Comment:
Previously, this would clone the entire `HashMap<>` for every single group message, where the key is a `Vec<u8>`